src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt

```diff
@@ -29,6 +34,34 @@ object MovementListener : Listener {
         }
     }
 
+    @EventHandler(ignoreCancelled = true)
+    fun onPlayerFalling(playerMoveEvent: PlayerMoveEvent) {
+        if(DeeperConfig.data.maxFallingDistance == -1f){
+            return
+        }
+
+        val (player) = playerMoveEvent
+        val actualFallDistance = player.getVehicleRecursive()?.fallDistance ?: player.fallDistance
+        if (actualFallDistance > DeeperConfig.data.maxFallingDistance
+                && !player.isGliding
+                && (player.gameMode == GameMode.SURVIVAL || player.gameMode == GameMode.ADVENTURE)) {
+            val previousTick = previousFallingDamageTick[player.uniqueId]
+
+            if (previousTick != null && currentServerTick - previousTick < player.maximumNoDamageTicks) {
+                return
+            }
+
+            // Always deal a minimum of 1 damage, else the first damage tick would deal (almost) no damage
+            val damageToDeal = (actualFallDistance - DeeperConfig.data.maxFallingDistance) * DeeperConfig.data.fallingDamageMultiplier + 1
+
+            player.damage(0.01) // Damage animation
+            player.health = (player.health - damageToDeal).coerceIn(0.0, player.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.value) // Ignores armor
+            previousFallingDamageTick[player.uniqueId] = currentServerTick
+        } else if (previousFallingDamageTick.containsKey(player.uniqueId)) {
```

Comment:
Should be able to do `player.uniqueId in previousFallingDamageTick` and `previousFallingDamageTick -= player.uniqueId`.

I don't care too much what you prefer, just letting you know ;p

Reply:
Those are some cool shorthand yeah, thanks!